includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -0,0 +1,252 @@
+		);
+	}
+
+	/**
+	 * Column cb.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_cb( $log ) {
+		return sprintf( '<input type="checkbox" name="log[]" value="%1$s" />', $log['log_id'] );
+	}
+
+	/**
+	 * Timestamp column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_timestamp( $log ) {
+		return esc_html( $log['timestamp'] );
+	}
+
+	/**
+	 * Level column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_level( $log ) {
+		$level_key = $log['level'];
+		$levels    = array(
+			'emergency' => _x( 'Emergency', 'Log level', 'woocommerce' ),
+			'alert'     => _x( 'Alert', 'Log level', 'woocommerce' ),
+			'critical'  => _x( 'Critical', 'Log level', 'woocommerce' ),
+			'error'     => _x( 'Error', 'Log level', 'woocommerce' ),
+			'warning'   => _x( 'Warning', 'Log level', 'woocommerce' ),
+			'notice'    => _x( 'Notice', 'Log level', 'woocommerce' ),
+			'info'      => _x( 'Info', 'Log level', 'woocommerce' ),
+			'debug'     => _x( 'Debug', 'Log level', 'woocommerce' ),
+		);
+
+		if ( isset( $levels[ $level_key ] ) ) {
+			return esc_html( $levels[ $level_key ] );
+		} else {
+			return '';
+		}
+	}
+
+	/**
+	 * Message column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_message( $log ) {
+		return esc_html( $log['message'] );
+	}
+
+	/**
+	 * Tag column.
+	 *
+	 * @param  array $log
+	 * @return string
+	 */
+	public function column_tag( $log ) {
+		return esc_html( $log['tag'] );
+	}
+
+	/**
+	 * Get bulk actions.
+	 *
+	 * @return array
+	 */
+	protected function get_bulk_actions() {
+		return array();
+	}
+
+	/**
+	 * Get a list of sortable columns.
+	 *
+	 * @return array
+	 */
+	protected function get_sortable_columns() {
+		return array(
+			'timestamp' => array( 'timestamp', true ),
+			'level'     => array( 'level', true ),
+			'tag'       => array( 'tag', true ),
+		);
+	}
+
+	/**
+	 * Prepare table list items.
+	 */
+	public function prepare_items() {
+		global $wpdb;
+
+		$per_page = apply_filters( 'woocommerce_status_log_items_per_page', 10 );
+		$columns  = $this->get_columns();
+		$hidden   = array();
+		$sortable = $this->get_sortable_columns();
+
+		// Column headers
+		$this->_column_headers = array( $columns, $hidden, $sortable );
+
+		$current_page = $this->get_pagenum();
+		if ( 1 < $current_page ) {
+			$offset = $per_page * ( $current_page - 1 );
+		} else {
+			$offset = 0;
+		}
+
+		$level_filter = '';
+		if ( ! empty( $_REQUEST['level'] ) ) {
+			$level_filter = $wpdb->prepare( 'AND level = %s', array( $_REQUEST['level'] ) );
+		}
+
+		$search = '';
+		if ( ! empty( $_REQUEST['s'] ) ) {
+			$search = "AND tag LIKE '%" . esc_sql( $wpdb->esc_like( wc_clean( $_REQUEST['s'] ) ) ) . "%' ";
+		}
+
+		if ( ! empty( $_REQUEST['orderby'] ) ) {
+			switch ( $_REQUEST['orderby'] ) {
+
+				// Level requires special case to order Emergency -> Debug
+				case 'level':
+					$order_by = 'CASE level '
```

Comment:
It makes the `level` key useless for ordering 😢 .

Solution: store level by an integer representation. The only disadvantages I see are that level strings must be translated on write or print and that the DB becomes more difficult to read for a human or third parties. If the latter is an issue, the level string can be stored in an additional field.